src/parser/core/modules/DisengageGcds.tsx

```diff
@@ -0,0 +1,53 @@
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {Action} from 'data/ACTIONS'
+import {Event} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import {SimpleStatistic, Statistics} from 'parser/core/modules/Statistics'
+import React from 'react'
+
+export class DisengageGcds extends Analyser {
+	static override handle = 'disengagegcds'
+
+	@dependency data!: Data
+	@dependency statistics!: Statistics
+
+	private disengageGcds: number = 0
+
+	// Pass in the GCD you want to track as an Action.
+	protected trackedAction: Action = this.data.actions.UNKNOWN
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('action').action(this.trackedAction.id), this.onUse)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	// Override this to suggest job-specific ways to handle downtime
+	protected addDisengageSuggestion() {
```

Comment:
Consider; I'm not liking that the stat maker is labelled as a suggestion, I feel it muddies the water about what the function does, but the module is small and simple enough that it can be read and understood quickly. 

I think it would be better if this was called addDisengageStatisitic() and a 2nd function addJobSpecificSuggestions() be added that is called onComplete() that can be overrode for suggestions, just to align it with other core module setups.

Reply:
Yes, that makes sense about terminology. In fact this the output from this is not a suggestion but a statistic, I've added an addJobSpecificStatistics() function that works in the same way as the one in the Combos module - setting it to true disables the default statistic output, setting it to false adds an additional statistic alongside the one in the core module.

I'm v new to using Typescript and unaware of best practices, advice very welcome on whether or not what I've done here is sane & in line with normal code style.